File: 118-pascals-triangle/118-pascals-triangle.go
func generate(numRows int) [][]int {
    
    pascalTriangle := make([][]int, numRows)
    for i := 0; i < numRows; i++ {
        pascalTriangle[i] = make([]int, i+1)
    }
    for i := 0; i < numRows; i++ {
        for j := 0; j < i + 1; j++ {
            if j == 0 || j + 1 == i + 1 {
                pascalTriangle[i][j] = 1
            } else {
                pascalTriangle[i][j] = pascalTriangle[i-1][j-1] + pascalTriangle[i-1][j]
            }
        }
    
    }
    return pascalTriangle   
}

File: 118-pascals-triangle/118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        
        tri = []
        for i in range(1, numRows + 1):
            row = [1] * i
            if i > 2:
                prev = 0
                s = 0
                for j, k in enumerate(tri[-1]):
                    s = prev + k
                    row[j] = s
                    prev = k
                    
                    
            tri.append(row)
        
        return tri